Implemented Infinite Scroll in Search Results

satyansh911 opened the pull request:
Hey @nafisreza, I have implemented Infinite Scroll which allows users to load 10 repositories first and then load them automatically as they scroll down.
Please review my work and let me know if any changes are required.
Thanks!

nafisreza reviewed (changes requested):
Thanks for your efforts. The infinite scroll loads more repos as I scroll. But after scrolling a few times, when I reach to the end it throws an error. Check it out.
<img width="950" height="596" alt="image" src="https://github.com/user-attachments/assets/1beeaab8-ca89-43f5-b754-fa96361f1004" />

satyansh911 commented:
Hey @nafisreza ! I wanted to give a quick overview of the changes I made in this PR:

- Fixed the infinite scroll so results now auto-load on page open and stop gracefully at the end.
- Moved GitHub search to a server API route using Octokit with GITHUB_TOKEN to prevent 403s and abuse detection issues.
- Deduplicated repos to resolve React “duplicate key” warnings.
- Added an IntersectionObserver for scroll-triggered loading, keeping a “Load more” button as a fallback.
- Capped results at 1,000 to respect GitHub API limits and provide a consistent UX.
- Improved error handling with friendly messages for rate limits or API abuse scenarios.
- Cleaned up linting and formatting, added proper types, removed unused code, and fixed exhaustive-deps warnings.

Overall, this should make scrolling, search, and rendering much smoother and safer.
Also, please try deploying from your side and do let me know if any other changes are required or if there are any roadblocks in deployment.
